engine/src/main/coffee/extensions/array.coffee

```diff
@@ -0,0 +1,87 @@
+# (C) Uri Wilensky. https://github.com/NetLogo/Tortoise
+
+# (Any) => Boolean
+isArray = (x) ->
+  x._type is "ext_array"
+
+notArrayException = (x, prim) ->
+  "Extension exception: not an array: #{JSON.stringify(x).replace(/,/g, ' ')} error while observer running #{prim}"
```

Comment:
You don't need to add the ` error while observer running #{prim}` part at the end of the error message.  The runtime engine should handle that for you.